src/database/postgres-db.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { DataSource } from 'typeorm';
+
+@Injectable()
+export class PostgresDbService {
+  constructor(private readonly dataSource: DataSource) {}
+
+  async connect(): Promise<void> {
+    console.log('Connected to PostgresSql');
+  }
+
+  async disconnect(): Promise<void> {
+    console.log('Disconnected from PostgresSql');
+  }
+
+  async insertOne(entity: any): Promise<void> {
+    await this.dataSource.manager.save(entity);
+  }
+
+  async findOne<T>(entityClass: { new (): T }, id: number): Promise<T | null> {
+    return this.dataSource.manager.findOne(entityClass, {
+      where: { id } as any,
+    });
+  }
```

Comment:
I don't understand this construction, do you want to set input type as output type?
You can set here the easy construction
Like I did:

```
 async findOne(table: MongooseModelsMapEnum, query: any): Promise<any> {
    const model = this.getModel(table);
    return model.findOne(query).lean();
  }

  private getModel(table: MongooseModelsMapEnum): mongoose.Model<any> {
    switch (table) {
      case MongooseModelsMapEnum.USER:
        return UserModel;

      default:
        break;
    }
  }
```

And only set input/output types easilly like:
async findOne<T>(entityClass: MongooseModelsMapEnum, id: number): Promise<T | null> {